central/cluster/datastore/datastore_impl_postgres_test.go

```diff
@@ -0,0 +1,156 @@
+	s.NoError(err)
+	s.nsDatastore = ds
+
+	s.mockCtrl = gomock.NewController(s.T())
+	s.netEntities = netEntitiesMocks.NewMockEntityDataStore(s.mockCtrl)
+	s.nodeDataStore = nodeMocks.NewMockGlobalDataStore(s.mockCtrl)
+	s.netFlows = netFlowsMocks.NewMockClusterDataStore(s.mockCtrl)
+
+	s.nodeDataStore.EXPECT().GetAllClusterNodeStores(gomock.Any(), gomock.Any()).AnyTimes().Return(nil, nil)
+	s.netEntities.EXPECT().RegisterCluster(gomock.Any(), gomock.Any()).AnyTimes()
+	clusterDS, err := New(clusterPostgres.New(s.ctx, s.db), clusterHealthPostgres.New(s.ctx, s.db), clusterPostgres.NewIndexer(s.db), nil, ds, nil, s.nodeDataStore, nil, nil, s.netFlows, s.netEntities, nil, nil, nil, nil, nil, nil, ranking.ClusterRanker(), nil)
+	s.NoError(err)
+	s.clusterDatastore = clusterDS
+}
+
+func (s *ClusterPostgresDataStoreTestSuite) TearDownSuite() {
+	nsPostgres.Destroy(s.ctx, s.db)
+	clusterPostgres.Destroy(s.ctx, s.db)
+	s.db.Close()
+	s.mockCtrl.Finish()
+	s.envIsolator.RestoreAll()
+}
+
+func (s *ClusterPostgresDataStoreTestSuite) TestSearchWithPostgres() {
+	ctx := sac.WithAllAccess(context.Background())
+
+	// Upsert cluster.
+	s.netFlows.EXPECT().CreateFlowStore(gomock.Any(), gomock.Any()).Return(netFlowsMocks.NewMockFlowDataStore(s.mockCtrl), nil)
+	c1ID, err := s.clusterDatastore.AddCluster(ctx, &storage.Cluster{
+		Name:               "c1",
+		Labels:             map[string]string{"env": "prod"},
+		MainImage:          mainImage,
+		CentralApiEndpoint: centralEndpoint,
+	})
+	s.NoError(err)
+
+	// Basic unscoped search.
+	results, err := s.clusterDatastore.Search(ctx, pkgSearch.EmptyQuery())
+	s.NoError(err)
+	s.Len(results, 1)
+
+	// Upsert cluster.
+	s.netFlows.EXPECT().CreateFlowStore(gomock.Any(), gomock.Any()).Return(netFlowsMocks.NewMockFlowDataStore(s.mockCtrl), nil)
+	c2ID, err := s.clusterDatastore.AddCluster(ctx, &storage.Cluster{
+		Name:               "c2",
+		Labels:             map[string]string{"env": "test"},
+		MainImage:          mainImage,
+		CentralApiEndpoint: centralEndpoint,
+	})
+	s.NoError(err)
+
+	// Basic unscoped search.
+	results, err = s.clusterDatastore.Search(ctx, pkgSearch.EmptyQuery())
+	s.NoError(err)
+	s.Len(results, 2)
+
+	ns1C1 := fixtures.GetNamespace(c1ID, "c1", "n1")
+	ns2C1 := fixtures.GetNamespace(c1ID, "c1", "n2")
+	ns1C2 := fixtures.GetNamespace(c2ID, "c2", "n1")
+
+	// Upsert namespace.
+	s.NoError(s.nsDatastore.AddNamespace(ctx, ns1C1))
+
+	// Basic unscoped search.
+	results, err = s.nsDatastore.Search(ctx, pkgSearch.EmptyQuery())
+	s.NoError(err)
+	s.Len(results, 1)
+
+	// Upsert.
+	s.NoError(s.nsDatastore.UpdateNamespace(ctx, ns2C1))
+	s.NoError(s.nsDatastore.UpdateNamespace(ctx, ns1C2))
+
+	// Basic unscoped search.
+	results, err = s.nsDatastore.Search(ctx, pkgSearch.EmptyQuery())
+	s.NoError(err)
+	s.Len(results, 3)
+
+	// Query cluster with namespace search field.
+	results, err = s.clusterDatastore.Search(ctx, pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.Namespace, "n2").ProtoQuery())
```

Comment:
Can you add some more complex queries here? Two properties on namespace checks with 1 property on cluster checked?